pkg/signing/signing.go

```diff
@@ -274,6 +277,50 @@ func getSigsPrimPart(fimg *sif.FileImage) (sigs []*sif.Descriptor, descr []*sif.
 	return
 }
 
+func getSigsAllPart(fimg *sif.FileImage) ([]*sif.Descriptor, []*sif.Descriptor, error) {
+	var dpart []*sif.Descriptor
+	var sigs []*sif.Descriptor
+	var err error
+
+	// Ensure theres a primary partition.
+	_, _, err = fimg.GetPartPrimSys()
+	if err != nil {
+		return nil, nil, fmt.Errorf("no primary partition found")
+	}
+
+	// Loop through all the partitions, (skipping DataSignatures)
+	// and collect all the signatures for a data partition.
+	for _, d := range fimg.DescrArr {
+		if !d.Used {
+			continue
+		}
+		// No need to verify a signature.
+		if d.Datatype == sif.DataSignature {
+			continue
+		}
+		s, _, err := fimg.GetLinkedDescrsByType(d.ID, sif.DataSignature)
+		if err != nil {
+			// If a partition is not signed, print a warning.
+			sylog.Warningf("No signature found for %s\n", datatypeStr(d.Datatype))
+			continue
+		}
+
+		// For every signature part, there will be a data partition.
+		for _, sp := range s {
+			sigs = append(sigs, sp)
+
+			partDescr := d
```

Comment:
I had to think for a second as to why you are doing this here...

This is Go telling you that you probably shouldn't be doing this in the first place.

As the descriptor array (fimg.DescrArr) is not going to change under you, and the elements of the array are readily available, it's probably a better idea to store indexes into the array than pointers to sif.Descriptor.

You might be better off returning a single slice, too:

```
type signatureLink {
   sigIndex  int  // the index of the descriptor with the signature
   dataIndex int  // the index of the descriptor of the signed data
}

func getAllSignatureLinks(fimg *sif.FileImage) ([]signatureLink, error) {
    // ...
    _, idxs, err := fimg.GetLinkedDescrsByType(d.ID, sif.DataSignature)
    // ...
    for sidx := range idxs {
       tbl = append(tbl, signatureLink{ sidx, didx })
    }
    // ...
    return tbl, nil
}
```